src/include/snapper/dialogs.rb

```diff
@@ -1233,6 +1018,196 @@ def ShowDialog
       deep_copy(ret)
     end
 
+
+    private
+
+    def ReallyAbort
+      Popup.ReallyAbort(true)
+    end
+
+
+    # Read settings dialog
+    # @return `abort if aborted and `next otherwise
+    def ReadDialog
+      return :abort if !Confirm.MustBeRoot
+
+      Wizard.RestoreHelp(Ops.get_string(@HELPS, "read", ""))
+      ret = Snapper.Init()
+      ret ? :next : :abort
+    end
+
+    def open_modify_dialog(content)
+      UI.OpenDialog(
+        Opt(:decorated),
+        HBox(
+          HSpacing(1),
+          VBox(
+            VSpacing(0.5),
+            HSpacing(65),
+            content,
+            VSpacing(0.5),
+            ButtonBox(
+              PushButton(Id(:ok), Label.OKButton),
+              PushButton(Id(:cancel), Label.CancelButton)
+            ),
+            VSpacing(0.5)
+          ),
+          HSpacing(1)
+        )
+      )
+    end
+
+    def snapshot_term(prefix, data)
+      data = deep_copy(data)
+      HBox(
+        HSpacing(),
+        Frame(
+          "",
+          HBox(
+            HSpacing(0.4),
+            VBox(
+              # text entry label
+              InputField(
+                Id(Ops.add(prefix, "description")),
+                Opt(:hstretch),
+                _("Description"),
+                Ops.get_string(data, "description", "")
+              ),
+              # text entry label
+              InputField(
+                Id(Ops.add(prefix, "userdata")),
+                Opt(:hstretch),
+                _("User data"),
+                Snapper.userdata_to_string(data["userdata"])
+              ),
+              Left(
+                ComboBox(
+                  Id(Ops.add(prefix, "cleanup")),
+                  Opt(:editable, :hstretch),
+                  # combo box label
+                  _("Cleanup algorithm"),
+                  cleanup_items(Ops.get_string(data, "cleanup", ""))
+                )
+              )
+            ),
+            HSpacing(0.4)
+          )
+        ),
+        HSpacing()
+      )
+    end
+
+    def update_snapshots
+      # busy popup message
+      Popup.ShowFeedback("", _("Reading list of snapshots..."))
+
+      Snapper.ReadSnapshots()
+
+      Popup.ClearFeedback
+
+      UI.ChangeWidget(Id(:snapshots_table), :Items, get_snapshot_items)
+      selected = UI.QueryWidget(Id(:snapshots_table), :SelectedItems) || []
+      enable_buttons([:modify, :show, :delete], selected.size == 1)
+      enable_buttons([:delete], selected.size >= 1)
```

Comment:
I see this snipper at least three times, so to be DRY, I recommend to move it to own method like `adjust_buttons` that query selected and adapt buttons according to it.